pandas/tests/indexes/test_old_base.py

```diff
@@ -213,7 +213,7 @@ def test_numeric_compat(self, simple_index):
             1 // idx
 
     def test_logical_compat(self, simple_index):
-        if simple_index.dtype in (object, "string"):
+        if simple_index.dtype in (object, "string", "datetime64[ns]"):
```

Comment:
Is the datetime skip required here? I am not overly familiar with this part of the code base but judging by the deleted comment I didn't think this needed to change

Reply:
I think we shouldn't skip datetime64 here. We need the separate branch for datetime64 because we get different error message: "datetime64 type does not support operation: '(any|all)'"

If we replace the error message 
https://github.com/pandas-dev/pandas/blob/6f39c4fa1fecec6fe447ec89bd1409d1427d1784/pandas/core/nanops.py#L521-L523
with the msg "does not support reduction '(any|all)'" 
we can combine these two checks. Can I do it in this PR or a new one?

Reply:
I suggest to align the error messages and assign `msg = “.* does not support operation .*“ `. I replaced `reduction` with `operation` and `operation:` with `operation` to unify error messages. Tests passed.